Compilation warnings

### Prerequisite Checklist

- [X] I searched for [existing issues](https://github.com/search?q=repo%3ASFML%2FSFML&type=issues) to prevent duplicates
- [X] I searched for [existing discussions on the forum](https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fen.sfml-dev.org) to prevent duplicates
- [X] I am here to report an issue and not to just ask a question or look for help (use the [forum](https://en.sfml-dev.org/forums/index.php#c3) or [Discord](https://discord.gg/nr4X7Fh) instead)

### Describe your issue here

I can produce warnings that with -Werror will halt the compilation from GLCheck.hpp and OpenGL_ES when compiling with -DSFML_OPENGL_ES=TRUE -DSFML_USE_DRM=FALSE

### Your Environment

- OS / distro / window manager: Debian 11
- SFML version: 2.6.0 (SHA1: 11b73743c42cf7ecd7c596ba83fdbf1150ffa96c)
- Compiler / toolchain: g++-10
- Special compiler / CMake flags: yes cmake -DWARNINGS_AS_ERRORS=OFF to allow compiling


### Steps to reproduce

```bash
mkdir build
cd build

cmake -DSFML_BUILD_AUDIO=FALSE -DSFML_BUILD_DOC=FALSE -DSFML_BUILD_EXAMPLES=FALSE -DSFML_BUILD_NETWORK=FALSE -DSFML_BUILD_TEST_SUITE=FALSE -DSFML_BUILD_WINDOW=TRUE -DSFML_OPENGL_ES=TRUE -DSFML_USE_DRM=FALSE -DBUILD_SHARED_LIBS=OFF -DWARNINGS_AS_ERRORS=OFF ..

make -j8
```

will produce errors:

```
In file included from src/SFML/Graphics/Texture.cpp:30:
src/SFML/Graphics/Texture.cpp: In member function ‘bool sf::Texture::create(unsigned int, unsigned int)’:
src/SFML/Graphics/Texture.cpp:210:108: warning: this condition has identical branches [-Wduplicated-branches]
  210 |     glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:211:108: warning: this condition has identical branches [-Wduplicated-branches]
  211 |     glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp: In member function ‘sf::Image sf::Texture::copyToImage() const’:
src/SFML/Graphics/Texture.cpp:349:43: warning: conversion to ‘GLsizei’ {aka ‘int’} from ‘unsigned int’ may change the sign of the result [-Wsign-conversion]
  349 |         glCheck(glReadPixels(0, 0, m_size.x, m_size.y, GL_RGBA, GL_UNSIGNED_BYTE, &pixels[0]));
      |                                    ~~~~~~~^
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:349:53: warning: conversion to ‘GLsizei’ {aka ‘int’} from ‘unsigned int’ may change the sign of the result [-Wsign-conversion]
  349 |         glCheck(glReadPixels(0, 0, m_size.x, m_size.y, GL_RGBA, GL_UNSIGNED_BYTE, &pixels[0]));
      |                                              ~~~~~~~^
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:352:63: warning: conversion to ‘GLuint’ {aka ‘unsigned int’} from ‘GLint’ {aka ‘int’} may change the sign of the result [-Wsign-conversion]
  352 |         glCheck(GLEXT_glBindFramebuffer(GLEXT_GL_FRAMEBUFFER, previousFrameBuffer));
      |                                                               ^~~~~~~~~~~~~~~~~~~
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp: In member function ‘void sf::Texture::setRepeated(bool)’:
src/SFML/Graphics/Texture.cpp:680:116: warning: this condition has identical branches [-Wduplicated-branches]
  680 |             glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:681:116: warning: this condition has identical branches [-Wduplicated-branches]
  681 |             glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~

In file included from src/SFML/Graphics/Texture.cpp:30:
src/SFML/Graphics/Texture.cpp: In member function ‘bool sf::Texture::create(unsigned int, unsigned int)’:
src/SFML/Graphics/Texture.cpp:210:108: warning: this condition has identical branches [-Wduplicated-branches]
  210 |     glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:211:108: warning: this condition has identical branches [-Wduplicated-branches]
  211 |     glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp: In member function ‘sf::Image sf::Texture::copyToImage() const’:
src/SFML/Graphics/Texture.cpp:349:43: warning: conversion to ‘GLsizei’ {aka ‘int’} from ‘unsigned int’ may change the sign of the result [-Wsign-conversion]
  349 |         glCheck(glReadPixels(0, 0, m_size.x, m_size.y, GL_RGBA, GL_UNSIGNED_BYTE, &pixels[0]));
      |                                    ~~~~~~~^
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:349:53: warning: conversion to ‘GLsizei’ {aka ‘int’} from ‘unsigned int’ may change the sign of the result [-Wsign-conversion]
  349 |         glCheck(glReadPixels(0, 0, m_size.x, m_size.y, GL_RGBA, GL_UNSIGNED_BYTE, &pixels[0]));
      |                                              ~~~~~~~^
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:352:63: warning: conversion to ‘GLuint’ {aka ‘unsigned int’} from ‘GLint’ {aka ‘int’} may change the sign of the result [-Wsign-conversion]
  352 |         glCheck(GLEXT_glBindFramebuffer(GLEXT_GL_FRAMEBUFFER, previousFrameBuffer));
      |                                                               ^~~~~~~~~~~~~~~~~~~
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp: In member function ‘void sf::Texture::setRepeated(bool)’:
src/SFML/Graphics/Texture.cpp:680:116: warning: this condition has identical branches [-Wduplicated-branches]
  680 |             glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
src/SFML/Graphics/Texture.cpp:681:116: warning: this condition has identical branches [-Wduplicated-branches]
  681 |             glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, m_isRepeated ? GL_REPEAT : (textureEdgeClamp ? GLEXT_GL_CLAMP_TO_EDGE : GLEXT_GL_CLAMP)));
src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro ‘glCheck’
   51 |     #define glCheck(expr) (expr)
      |                            ^~~~
```

### Expected behavior

No warnings since compiled with -Werror

### Actual behavior

Undesired warnings

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -16,6 +16,7 @@ jobs:
         - { name: Windows VS2022 Clang, os: windows-2022, flags: -T ClangCL }
         - { name: Linux GCC,      os: ubuntu-20.04  }
         - { name: Linux Clang,    os: ubuntu-20.04, flags: -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ }
+        - { name: Linux GCC OpenGL ES, os: ubuntu-20.04, flags: -DSFML_OPENGL_ES=ON }
         - { name: MacOS XCode,    os: macos-11   }
         config:
         - { name: Shared,       flags: -DBUILD_SHARED_LIBS=TRUE }

--- a/changelog.md
+++ b/changelog.md
@@ -24,6 +24,7 @@ Also available on the website: https://www.sfml-dev.org/changelog.php#sfml-2.6.1
 **Bugfixes**
 
 -   Ensure OpenGL extensions are loaded before querying maximum texture size (#2603)
+-   Fix warnings in Linux OpenGL ES codepaths (#2747)
 
 ### Audio
 

--- a/src/SFML/Graphics/RenderTextureImplFBO.cpp
+++ b/src/SFML/Graphics/RenderTextureImplFBO.cpp
@@ -387,7 +387,7 @@ bool RenderTextureImplFBO::create(unsigned int width, unsigned int height, unsig
     if (createFrameBuffer())
     {
         // Restore previously bound framebuffer
-        glCheck(GLEXT_glBindFramebuffer(GLEXT_GL_FRAMEBUFFER, frameBuffer));
+        glCheck(GLEXT_glBindFramebuffer(GLEXT_GL_FRAMEBUFFER, static_cast<GLuint>(frameBuffer)));
 
         return true;
     }

--- a/src/SFML/Graphics/Texture.cpp
+++ b/src/SFML/Graphics/Texture.cpp
@@ -38,6 +38,10 @@
 #include <cstring>
 #include <climits>
 
+#if defined(__GNUC__)
+    #pragma GCC diagnostic ignored "-Wduplicated-branches"
+#endif
+
 
 namespace
 {
@@ -346,10 +350,16 @@ Image Texture::copyToImage() const
 
         glCheck(GLEXT_glBindFramebuffer(GLEXT_GL_FRAMEBUFFER, frameBuffer));
         glCheck(GLEXT_glFramebufferTexture2D(GLEXT_GL_FRAMEBUFFER, GLEXT_GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, m_texture, 0));
-        glCheck(glReadPixels(0, 0, m_size.x, m_size.y, GL_RGBA, GL_UNSIGNED_BYTE, &pixels[0]));
+        glCheck(glReadPixels(0,
+                             0,
+                             static_cast<GLsizei>(m_size.x),
+                             static_cast<GLsizei>(m_size.y),
+                             GL_RGBA,
+                             GL_UNSIGNED_BYTE,
+                             &pixels[0]));
         glCheck(GLEXT_glDeleteFramebuffers(1, &frameBuffer));
 
-        glCheck(GLEXT_glBindFramebuffer(GLEXT_GL_FRAMEBUFFER, previousFrameBuffer));
+        glCheck(GLEXT_glBindFramebuffer(GLEXT_GL_FRAMEBUFFER, static_cast<GLuint>(previousFrameBuffer)));
 
         if (m_pixelsFlipped)
         {

